### PR TITLE
Fix issues where values of `0` were being not sent to server

### PR DESCRIPTION
**Pull request by sgnn7**
This was a particular issue for count field since it starts at 0 and is
falsey.

https://phabricator.endlessm.com/T20993

### CODE REVIEW

**Comment by sgnn7** [comment]
@wjt Hrmm... thought it was `None`? Maybe we should wait for @ramcq since he was working on it originally.

**Comment by ramcq** [comment]
Hmm. It used to be that empty strings would be replaced with "unknown" but now that's only done for mandatory strings. I think we should mark 'serial' and 'metrics_environment' as mandatory - all we're trying to do is permit 'mac_hash' to be omitted. 'dualboot', 'live', and 'metrics_enabled' are boolean so this check is good to have! will push an additional patch to add serial and metrics_environment as mandatory.

File: eos-phone-home [hunk 1]
```diff
@@ -85,7 +85,8 @@ class PhoneHome(object):
 
     # variables which should be replaced with "unknown" rather
     # than omitted if the get function returns None
-    MANDATORY_STRINGS = ('image', 'vendor', 'product', 'release')
+    MANDATORY_STRINGS = ('image', 'vendor', 'product', 'release', 'serial',
```
[the remark this answers is not in the document]
Comment (sgnn7) [reply eos-phone-home]: @ramcq Sure though I wouldn't expect many cases of this failing and thus including "unknown"

[the remark this answers is not in the document]
Comment (ramcq) [reply eos-phone-home]: Well, just trying to reinstate the behaviour we had before. serial could be blank thanks to clever manufacturers. metrics_environment would have to be pretty messed up to be '' though. :)

**Pull request merged**